File: m3tile.py
sum = 1
F = [1] * 16

for i in range(1, 16):
    F[i] = sum * 2 + F[i-1]
    sum += F[i]

while True:
    n = int(input())
    if n == -1:
        break
    if n % 2 == 1:
        print(0)
    else:
        print(F[n // 2])

File: m3tile.cpp
#include <stdio.h>

int main(){
    int * f = new int[16];
    f[0] = 1;
    f[1] = 3;
    int sum = 4;
    for(int i = 2;i<16;i++){
        f[i] = sum * 2 + f[i-1];
        sum += f[i];
    }

    int n;
    while(true){
        scanf("%d", &n);
        if(n < 0)
            break;
        if(n % 2)
            printf("0\n");
        else
            printf("%d\n", f[n >> 1]);
    }
    return 0;
}
